specification/basic_concepts/basic_concepts.tex

```diff
@@ -29,38 +29,30 @@ \chapter{Basic concepts}\label{sec:basic_concepts}
 A DSDL description can be used to automatically generate the serialization/deserialization code
 for every defined data structure in a particular programming language.
 DSDL ensures that the worst case memory footprint and computational complexity per data type
-are constant and easily predictable, which is paramount for hard real-time and safety-critical applications.
+are constant and easily predictable.
 
 On top of the standard data types, UAVCAN defines a set of standard high-level functions including:
-node health monitoring, network discovery, time synchronization, firmware update,
+node health monitoring, node discovery, time synchronization, firmware update,
 plug-and-play node support, and more.
 For more information see chapter \ref{sec:application_layer}.
 
 Serialized message and service data structures are exchanged by means of the transport
 layer (chapter \ref{sec:transport_layer}),
-which implements automatic decomposition/reassembly of long transfers into/from several transport
+which implements automatic decomposition/reassembly of long transfers from/into several transport
 frames\footnote{Here and elsewhere, a \emph{transport frame} means a block of data
 that can be atomically exchanged over the network, e.g., a CAN frame.},
 allowing nodes to exchange data structures of arbitrary size.
 
 \begin{figure}[hbt]
     \centering
-    \setlength\arrayrulewidth{1pt}  % https://tex.stackexchange.com/a/256732/132781
-    \begin{tabularx}{0.8\textwidth}{|X|X|X|X|}
-        \hline
-        \multicolumn{4}{|c|}{Application} \\
-        \hline
-
-        & Standard functions & \multicolumn{2}{c|}{\multirow{3}{*}{Vendor-specific data types (optional)}} \\\cline{2-2}
-        \multicolumn{2}{|c|}{\multirow{2}{*}{Standard data types}} & \multicolumn{2}{c|}{} \\
-        \multicolumn{2}{|c|}{} & \multicolumn{2}{c|}{} \\
-
-        \hline
-        \multicolumn{4}{|c|}{Serialization} \\
-        \hline
-        \multicolumn{4}{|c|}{Transport layer} \\
+    \begin{tabular}{|c|c|l|c|l|}
         \hline
-    \end{tabularx}
+        \multicolumn{5}{|c|}{Application} \\ \hline
+        \multicolumn{1}{|l|}{\cellcolor[HTML]{FCE5CE}} & \cellcolor[HTML]{B5D8A6}{\color[HTML]{000000} Required Functions} & \cellcolor[HTML]{DD7E6B} & \cellcolor[HTML]{FFE593}{\color[HTML]{333333} Standard Functions} & \cellcolor[HTML]{D6A5BD} \\ \cline{2-2} \cline{4-4}
```

Comment:
Inconsistent capitalization: Title Case is not used in this diagram excepting the two cells you added: `Required Functions` and `Standard Functions`. I suggest replacing them with `Required functions` and `Standard functions`.

Also, at this point we might want to consider adding `Custom functions` for extra consistency?

Also also, not sure what went wrong, but the table is rendered black-and-white, despite the cell color directives. Perhaps they should be removed, seeing as they don't work? We can get back to visual refinements later when the actual content is finished, if desired.

Reply:
...and this is where I admit I have no friggin idea how to write TeX :/ I used an online TeX table generator for this. I'll fix the issues you pointed out but I don't know why the colouring hasn't worked.